### DISCUSSION TITLE
keyball39 ジャンパのはんだ失敗対応とその他不明点について

### DISCUSSION THREAD

**Discussion by sasaken-jmz** [Yowkees/keyball#272] [Q&A]
お疲れ様です。keyball39を組み立てている者です。
その中で問題と不明点が発生したため、ご教示願います。
右手ball有側となります。
自作初心者でかなりトラブルが多い状態で恐縮ですが、よろしくお願いいたします。

**問題：まずはこちらを解決する必要がある認識**
・ジャンパのはんだづけができない状態になってしまった。
　元はトラックボールが認識されなかったことから、ジャンパのはんだのやり直しを実施していた。
　その結果部材が剥がれ、はんだづけができない状態になった。
　過去Discussionより銅線を用いて何処かをショートさせれば良さそうなのは認識しているが、
　回路図を見ても勉強不足のため該当箇所がわからない。

以下該当ジャンパについての画像を添付します。
赤丸が該当部分です。

**不明点：ジャンパの影響かもしれないため、上記を解決してから考える方が良いのではという認識**
・画像青丸
　スイッチを押下しても文字が入力されない（REMAPでアサインはできている）
・画像黄色
　どちらか片方のスイッチを押下すると、両方の文字が入力される

以上、よろしくお願いいたします。
![IMG_2119](https://github.com/Yowkees/keyball/assets/56680587/ce4b09c0-c562-4b73-a726-cc4eea88e2d2)

**Comment by sasaken-jmz** [comment]
補足です。
勉強不足ではありますが、テスターは所持しているため、導通確認は実施できる状態です。
原因の切り分けとしてどのような導通確認を実施すればよいかなどもございましたらいただけたらと思います。
重ねてになりますがよろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
まずトラックボールのジャンパですが、ちょっと距離が遠いですが、赤点どうしをエナメル線等で接続してください。これでジャンパしたのと同じ回路になります。

キーについては、1列すべて不具合がある場合はProMicro裏の半田付けで、隣の端子とショートしている原因が一番可能性が高いと思います。ProMicro裏はスルーホールがジグザグに配置されていて隣と距離が近くショートしやすいです。

画像の緑点で示した３点が不具合のあるキーに該当します。
テスターをお持ちでしたら、緑点が周辺のスルーホールとショートしていないか確認いただき、ショートしているものは半田ごてで再加熱して切り離してください。
ProMicro裏の半田付けは念のため、全体的に見直してみてください。

以上、お手数ですがご対応お願いいたします。

![KB39_keyfix](https://github.com/Yowkees/keyball/assets/84209772/3887c72f-a069-4344-b44b-870755411c1d)

**Comment by sasaken-jmz** [comment]
ありがとうございます。
ご教示いただいた方法でトラックボールの方は動作するようになりました。
本当にありがとうございます。

一方、キーの反応についてですが、ご指摘いただいたとおりProMico側でショートしている部分がありました。
添付画像の青の部分がショートしている状態であり、はんだを吸い取りましたが、状況が変わらなかったため、
部材を取り払いました。しかしながら、以前ショートしている状態であり、表面にはんだがまだうっすらと残っているのか、
それとも内部？を変形させてしまったのではないかと考えております。

考えられる原因と対処についてご教示いただけませんでしょうか。
お手数ですがよろしくお願いいたします。
![IMG_2153](https://github.com/Yowkees/keyball/assets/56680587/6748fbcb-7f1a-4ec5-988c-7a6138af7f26)
![IMG_2154](https://github.com/Yowkees/keyball/assets/56680587/c654c171-dbb7-4049-902d-7714ffa339ba)

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

ご連絡いただいた青丸の２点ですが、反対側のまだ付いている12Pピンソケットの下にも同じような配置でありますので、もう一方の12Pピンソケットも除去して確認する必要があります。

もし除去が難しいようでしたら、ミドル基板単体や、ミドル基板と半田付け部品のやり直しセットを販売することも可能です。その場合、配送先等のお話をしたいので、白銀ラボウェブショップのコンタクトからご連絡いただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Comment by sasaken-jmz** [comment]
ありがとうございます。
ご指摘通り逆側の12ピンソケットを除去したところ、元々ショートしていた箇所については解消されました。
逆側のピンソケット周りで不具合があったということでしょうか。

また一方で、画像青でマークした部分は最初から繋がっているように見受けられるのですが、
こちらお間違いないでしょうか。（今まで見落としていたかもしれませんが、再度一通り通電確認を実施したところ、ショートしておりました。）

上記2点認識間違いがなければ再度組み立て直しをしたいと思います。

やり直しセットのご案内ありがとうございます。
上記について回答をいただいたのちに検討したいと思います。
よろしくお願いいたします。
![IMG_2157](https://github.com/Yowkees/keyball/assets/56680587/95dcc4f3-94dc-442a-95f7-9e08fc31f0d3)

**Comment by Yowkees** [comment]
新しくご連絡いただいた青点２点はGNDですのでショートしていて大丈夫です。
下記リポジトリに回路図や基板データを公開しておりますので、KiCAD等フリーソフトで表示していただくと参考になると思います。
https://github.com/Yowkees/keyball/tree/main/keyball39/design_data

昨日回答させていただいた青点２点のショートが解消されたということで、このまま組み上げていただければ、全キー動くと思いますよ。
おそらく反対側ピンソケットと基板の隙間でショートしていたと思われます。
ぜひあと少し頑張ってください！

**Comment by sasaken-jmz** [comment]
ありがとうございます。
大変恐縮なのですが、追加で1点質問がございます。（おそらく最後だと信じています。。。）

添付画像の青丸部分がショートしているようで、該当スイッチ部分が押しっぱなしの判定になってしまいます。
こちらについて原因・対処法をご教示願います。
回路図などを参照してみたつもりですが、解決せずでして。。。

以上、お願いいたします。
![IMG_2158](https://github.com/Yowkees/keyball/assets/56680587/0d0e667f-ee00-47a6-ac02-0a3724a69197)

**Comment by sasaken-jmz** [comment]
上記の件、解決いたしました！
promicroにいくつかショートを確認したので、デバッグを行ったところ、解決いたしました。
何度もお助けいただき、本当にありがとうございました。
また困ったことがあれば連絡させていただく可能性がございますが、どうかよろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡が遅くなり申し訳ありませんでした。
ご自身で解決いただけたということで、誠にありがとうございました。
ぜひKeyballをご愛用いただければ幸いです。
今後ともどうぞよろしくお願いいたします。

**Comment by sasaken-jmz** [comment]
さまざまなご指摘いただきありがとうございました！愛用いたします！！！
